Using the latest cached version of the dataset since flwrlabs/pacs couldn't be found on the Hugging Face Hub
Found the latest cached dataset configuration 'default' at /home/apiasecz/.cache/huggingface/datasets/flwrlabs___pacs/default/0.0.0/394113073258ead631f617d2e13bb377c0715c4b (last modified on Wed Jan 29 16:59:35 2025).
Using the latest cached version of the dataset since flwrlabs/pacs couldn't be found on the Hugging Face Hub
Found the latest cached dataset configuration 'default' at /home/apiasecz/.cache/huggingface/datasets/flwrlabs___pacs/default/0.0.0/394113073258ead631f617d2e13bb377c0715c4b (last modified on Wed Jan 29 16:59:35 2025).
Using the latest cached version of the dataset since flwrlabs/pacs couldn't be found on the Hugging Face Hub
Found the latest cached dataset configuration 'default' at /home/apiasecz/.cache/huggingface/datasets/flwrlabs___pacs/default/0.0.0/394113073258ead631f617d2e13bb377c0715c4b (last modified on Wed Jan 29 16:59:35 2025).
Device: cuda
CUDA Version: 12.1
cuDNN Version: 8902
PyTorch Version: 2.3.0

Model used: PACSCNN
Initial dataset creation
Initial dataset creation
Initial loss: 146.21334838867188
<__main__.Policy object at 0x147469c856d0>
146.21334838867188
Loss historical diff: 0.0; Loss Difference: 0.0, Decision: 0
Round 1 took 4.862464666366577s, results:
{'t': 0, 'current_accuracy': 0.21484375, 'current_loss': 139.06764221191406, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['sketch']}
Loss historical diff: 2.288818359375e-05; Loss Difference: 0.0, Decision: 0
Round 2 took 4.832356214523315s, results:
{'t': 1, 'current_accuracy': 0.21484375, 'current_loss': 139.06766510009766, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['sketch']}
Loss historical diff: 0.0; Loss Difference: 0.0, Decision: 0
Round 3 took 4.58924412727356s, results:
{'t': 2, 'current_accuracy': 0.21484375, 'current_loss': 139.06764221191406, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['sketch']}
Loss historical diff: 1.52587890625e-05; Loss Difference: 0.0, Decision: 0
Round 4 took 5.424726247787476s, results:
{'t': 3, 'current_accuracy': 0.21484375, 'current_loss': 139.06765747070312, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['sketch']}
Loss historical diff: 0.0; Loss Difference: 0.0, Decision: 0
Round 5 took 5.369267463684082s, results:
{'t': 4, 'current_accuracy': 0.21484375, 'current_loss': 139.06764221191406, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['sketch']}
Loss historical diff: 7.62939453125e-06; Loss Difference: 0.0, Decision: 0
Round 6 took 5.515431642532349s, results:
{'t': 5, 'current_accuracy': 0.21484375, 'current_loss': 139.0676498413086, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['sketch']}
Loss historical diff: 1.52587890625e-05; Loss Difference: 0.0, Decision: 0
Round 7 took 5.362181663513184s, results:
{'t': 6, 'current_accuracy': 0.21484375, 'current_loss': 139.06765747070312, 'train_loss': None, 'decision': 0, 'drift_rate': 0.0, 'target_domains': ['sketch']}
Loss historical diff: 0.0; Loss Difference: 0.0, Decision: 1
Traceback (most recent call last):
  File "/home/apiasecz/programming/concept_drift_optimal_adaptation_FL/src/execute/evaluate_policy.py", line 753, in <module>
    main()
  File "/home/apiasecz/programming/concept_drift_optimal_adaptation_FL/src/execute/evaluate_policy.py", line 733, in main
    evaluate_policy_under_drift(
  File "/home/apiasecz/programming/concept_drift_optimal_adaptation_FL/src/execute/evaluate_policy.py", line 554, in evaluate_policy_under_drift
    update_loss = agent_train.update_steps(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/programming/Federated_Learning_Base_Toolkit_torch/fl_toolkit/federated/client.py", line 202, in update_steps
    return self.model.update_steps(self.data_loader, optimizer, loss_fn, num_updates, verbose)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/apiasecz/programming/Federated_Learning_Base_Toolkit_torch/fl_toolkit/models/base_model.py", line 77, in update_steps
    loss.backward()
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/_tensor.py", line 525, in backward
    torch.autograd.backward(
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/autograd/__init__.py", line 267, in backward
    _engine_run_backward(
  File "/home/apiasecz/.conda/envs/cog_fl_llm_env/lib/python3.12/site-packages/torch/autograd/graph.py", line 744, in _engine_run_backward
    return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: cuDNN error: CUDNN_STATUS_NOT_INITIALIZED
